pkg/tasks/tasks.go

```diff
@@ -21,43 +21,60 @@ import (
 	"k8s.io/klog/v2"
 )
 
+// TaskRunner manages lists of tasks. Through the RunAll method tasks are
+// executed, the lists sequentially, each list of tasks concurrently.
 type TaskRunner struct {
-	client *client.Client
-	tasks  []*TaskSpec
+	client    *client.Client
+	taskLists [][]*TaskSpec
```

Comment:
What is the reason for changing the type here?

Reply:
Does the [commit message for this change](https://github.com/openshift/cluster-monitoring-operator/pull/1193/commits/d6d1510c5253e1ec2cdeedf3b2d5bbbb8345697f) make things clearer?

Part of this PR is to extend `TaskRunner` to process multiple groups of tasks as a way to enforce some order.

Reply:
Ah yes, that makes sense. To make sure this is reflected in the codebase, can we introduce a new abstraction (e.g. `TaskGroup`) where we can document this as part of the godoc?

Reply:
This is definitely just optional and if it is too big of a change it's fine to keep the existing solution.

Reply:
I agree with @fpetkovski that it might be good to introduce another struct like `TaskGroup` for clarity.